templates/base.html.twig

```diff
@@ -80,26 +80,48 @@
                                 {% endif %}
 
                                 <li class="nav-item dropdown">
-                                    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false" id="locales">
-                                        <i class="fa fa-globe" aria-hidden="true"></i>
+                                    <a href="#" class="nav-link dropdown-toggle" data-toggle="modal" data-target="#locale-selector-modal" role="button" id="locales">
+                                        <i class="fa fa-solid fa-globe" aria-hidden="true"></i>
                                         <span class="caret"></span>
                                         <span class="sr-only">{{ 'menu.choose_language'|trans }}</span>
                                     </a>
-                                    <ul class="dropdown-menu locales" role="menu" aria-labelledby="locales">
-                                        {% for locale in locales() %}
-                                            <li {% if app.request.locale == locale.code %}aria-checked="true" class="dropdown-item active"{% else %}class="dropdown-item" aria-checked="false"{% endif %} role="menuitem">
-                                                <a href="{{ path(app.request.get('_route', 'blog_index'), app.request.get('_route_params', [])|merge({_locale: locale.code})) }}">{{ locale.name|capitalize }}
-                                                    <small>{{ locale.code[0:2] }}</small>
-                                                </a>
-                                            </li>
-                                        {% endfor %}
-                                    </ul>
                                 </li>
                             </ul>
                         </div>
                     </div>
                 </nav>
             </header>
+
+            <div id="locale-selector-modal" class="modal" tabindex="-1" aria-hidden="true">
+                <div class="modal-lg modal-dialog modal-dialog-centered">
+                    <div class="modal-content">
+                        <div class="modal-header">
+                            <h5 class="modal-title">
+                                {# display this title always in English to make the language selector easier to use #}
```

Comment:
you should put `lang="en"` on that element then, as it is not in the language defined in the `<html>` tag